types/seamless-immutable/index.d.ts

```diff
@@ -63,7 +63,9 @@ declare namespace SeamlessImmutable {
         getIn(propertyPath: string[]): Immutable<any>;
         getIn<TValue>(propertyPath: string[], defaultValue: TValue): Immutable<TValue>;
 
-        asMutable(opts?: AsMutableOptions): T;
+        asMutable(opts?: AsMutableOptions<true>): T;
+        asMutable(opts?: AsMutableOptions<false>): { [K in keyof T]: Immutable<T[K]> };
+        asMutable(opts?: AsMutableOptions): T | { [K in keyof T]: Immutable<T[K]> };
```

Comment:
Thanks, I'll make that change tonight.

Reply:
I wanted to allow AsMutableOptions<boolean> so for both arrays and objects I've removed the '?' from the true and boolean overloads. To help visually I've moved the optional parameter overload to be first.

I've also added some tests. I wasn't sure how to address Immutable<User> seems to be assignable to User so that case isn't covered.